content-test/addon/MetadataParser.test.js

```diff
@@ -0,0 +1,72 @@
+/* globals assert, require */
+"use strict";
+const createMetadataParser = require("inject!addon/MetadataParser");
+const TEST_HTML = `<html>
+  <head>
+    <title>Test Page</title>
+    <link rel="shortcut icon" type="image/x-icon" href="//www.foo.com/icon.png" />
+    <meta property="og:image" content="//www.foo.com/image.png" />
+  </head>
+ </html>`;
+
+describe("MetadataParser", () => {
+  let metadataParser;
+  function setup() {
+    const mockPreviewProvider = {
+      processLinks(link) {
+        return link.map(link => Object.assign({}, link, {cache_key: link.url, places_url: link.url}));
+      }
+    };
+    const {MetadataParser} = createMetadataParser({
+      "common/vendor.bundle": {
+        PageMetadataParser: {getMetadata(doc) {return doc;}},
+        url: {resolve(url) {return url;}}
+      },
+      "chrome": {
+        Cc: {"@mozilla.org/xmlextras/domparser;1": {createInstance: () => new DOMParser()}},
+        Ci: {nsIDOMParser: {}}
+      }
+    });
+    metadataParser = new MetadataParser(mockPreviewProvider);
+  }
+
+  beforeEach(() => setup());
+
+  it("should get the document object", () => {
+    const result = metadataParser._getDocumentObject(TEST_HTML);
+    assert.ok(result);
+    assert.equal(result.querySelector("title").textContent, "Test Page");
+  });
+
+  it("should format the data properly", () => {
+    let originalUrl = "https://www.foo.com";
+    let incomingData = {
+      "description": "Some description",
+      "icon_url": "//www.foo.com/icon.png",
+      "image_url": "//www.foo.com/image.png",
+      "title": "Some title",
+      "url": "https:///"
+    };
+    let formattedData = metadataParser._formatData(incomingData, originalUrl)[0];
```

Comment:
If we remove the dependency on processLinks then we don't get an array back (since we only expect a single entry anyways) so you can remove all the [0] indexing.